services/web-server/test/third_party_test.js

```diff
@@ -0,0 +1,478 @@
+const assert = require('assert');
+const taskcluster = require('taskcluster-client');
+const testing = require('taskcluster-lib-testing');
+const debug = require('debug')('third_party_test');
+const request = require('superagent');
+const moment = require('moment');
+const helper = require('./helper');
+const tryCatch = require('../src/utils/tryCatch');
+
+helper.secrets.mockSuite(testing.suiteName(), ['taskcluster'], function(mock, skipping) {
+  helper.withEntities(mock, skipping);
+  helper.withFakeAuth(mock, skipping);
+  helper.withServer(mock, skipping);
+
+  const url = path => `http://127.0.0.1:${helper.serverPort}${path}`;
+  const getQuery = (url, sep='?') => {
+    const qmark = url.indexOf(sep);
+    if (qmark === -1) {
+      return new URLSearchParams();
+    }
+    return new URLSearchParams(url.slice(qmark + 1));
+  };
+
+  suite('unit', function() {
+    test('authorization endpoint redirects to the root page if user is not logged in', async function() {
```

Comment:
Is this the intent?  My hunch would be that it would redirect to `/third-party` page in the UI which would ask the user to click the "sign in" button and not display the AuthConsent form until that's done.